signac/core/validators.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Validators for SyncedCollection API."""
+import warnings
+from collections.abc import Mapping
+from collections.abc import Sequence
+
+from ..errors import InvalidKeyError
+from ..errors import KeyTypeError
+
+try:
+    import numpy
+    NUMPY = True
+except ImportError:
+    NUMPY = False
+
+
+def NoDotInKey(data):
+    """Raise an exception if there is a dot (``.``) in a mapping's key.
+
+    Parameters
+    ----------
+    data
+        Data to validate.
+
+    Returns
+    -------
+    Validated data.
```

Comment:
You need to change all validators to not return the data anymore. I didn't comment on this in the other files since it's everywhere and will be easier for you to just go through and delete all instances of `data = validator.validate(data)` or whatever. Removing the returns from these two validators should automatically cause your tests to fail until you fix this everywhere, so that will be easy enough to check.